src/bot.ts

```diff
@@ -1005,6 +1005,20 @@ export class DiscordBot {
                     formatted_body: result.formattedBody,
                     msgtype: result.msgtype,
                 };
+                if (msg.reference) {
+                    const storeEvent = await this.store.Get(DbEvent, {discord_id: msg.reference?.messageID})
+                    if (storeEvent && storeEvent.Result)
+                    {
+                        while(storeEvent.Next())
+                        {
+                            sendContent["m.relates_to"] = {
```

Comment:
`m.relates_to` is also set below with an override like this. The object structures seem incompatible but I think we should consider one of the following:

1. Merge the `m.relates_to` objects.
2. Put a comment explaining that edit relations will override reference relations and why that is correct.